src/leveldbwrapper.h

```diff
@@ -117,9 +145,11 @@ class CLevelDBWrapper
     template <typename K, typename V>
     bool Write(const K& key, const V& value, bool fSync = false) throw(leveldb_error)
     {
-        CLevelDBBatch batch;
-        batch.Write(key, value);
-        return WriteBatch(batch, fSync);
+        CLevelDBBatch* batch = GetBatch();
+        batch->Write(key, value);
+        bool written = WriteBatch(batch, fSync);
+        delete batch;
```

Comment:
Sounds good to me, though honestly I'm hoping to obviate the heap allocations entirely by absorbing `CLevelDBBatch` into `CLevelDBWrapper`... this is pending an okay from @sipa @gmaxwell.